src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,420 @@
+            if (!string.IsNullOrEmpty(id))
+            {
+                TimeZoneInfo? defaultTimeZone = GetTimeZone(id, id);
+
+                if (defaultTimeZone != null)
+                {
+                    return defaultTimeZone;
+                }
+            }
+
+            return Utc;
+        }
+
+        private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachineCore(string id, out TimeZoneInfo? value, out Exception? e)
+        {
+
+            value = id == LocalId ? GetLocalTimeZoneCore() : GetTimeZone(id, id);
+
+            if (value == null)
+            {
+                e = new InvalidTimeZoneException(SR.Format(SR.InvalidTimeZone_InvalidFileData, id, AndroidTzDataInstance.GetTimeZoneDirectory() + TimeZoneFileName));
+                return TimeZoneInfoResult.TimeZoneNotFoundException;
+            }
+
+            e = null;
+            return TimeZoneInfoResult.Success;
+        }
+
+        private static string[] GetTimeZoneIds()
+        {
+            return AndroidTzDataInstance.GetTimeZoneIds();
+        }
+
+        /*
+        * Android v4.3 Timezone support infrastructure.
+        *
+        * Android tzdata files are found in the format of
+        * Header <Beginning of Entry Index> Entry Entry Entry ... Entry <Beginning of Data Index> <TZDATA>
+        *
+        * https://github.com/aosp-mirror/platform_bionic/blob/master/libc/tzcode/bionic.cpp
+        *
+        * The header (24 bytes) contains the following information
+        * signature - 12 bytes of the form "tzdata2012f\0" where 2012f is subject to change
+        * index offset - 4 bytes that denotes the offset at which the index of the tzdata file starts
+        * data offset - 4 bytes that denotes the offset at which the data of the tzdata file starts
+        * final offset - 4 bytes that used to denote the final offset, which we don't use but will note.
+        *
+        * Each Data Entry (52 bytes) can be used to generate a TimeZoneInfo and contain the following information
+        * id - 40 bytes that contain the id of the time zone data entry timezone<id>
+        * byte offset - 4 bytes that denote the offset from the data offset timezone<id> data can be found
+        * length - 4 bytes that denote the length of the data for timezone<id>
+        * unused - 4 bytes that used to be raw GMT offset, but now is always 0 since tzdata2014f (L).
+        *
+        * This is needed in order to read Android v4.3 tzdata files.
+        *
+        * Android 10+ moved the up-to-date tzdata location to a module updatable via the Google Play Store and the
+        * database location changed (https://source.android.com/devices/architecture/modular-system/runtime#time-zone-data-interactions)
+        * The older locations still exist (at least the `/system/usr/share/zoneinfo` one) but they won't be updated.
+        */
+        private sealed class AndroidTzData
+        {
+            private unsafe struct AndroidTzDataHeader
```

Comment:
```suggestion
            private struct AndroidTzDataHeader
```

Reply:
`error CS8340: Instance fields of readonly structs must be readonly.`
`/Users/mdhwang/runtime_droid/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs(314,17): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)`

I went ahead and removed this struct altogether, using out parameters for indexOffset and dataOffset in `LoadTzFile` and `LoadHeader`